bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -291,6 +381,104 @@ where
 			[GlobalConsensus(network), AccountKey20 { network: None, key: token.into() }],
 		)
 	}
+
+	/// Constructs an XCM message destined for AssetHub that withdraws assets from the sovereign
+	/// account of the Gateway contract and either deposits those assets into a recipient account or
+	/// forwards the assets to another parachain.
+	fn convert_send_native_token(
+		message_id: H256,
+		chain_id: u64,
+		token_id: TokenId,
+		destination: Destination,
+		amount: u128,
+		asset_hub_fee: u128,
+	) -> Result<(Xcm<()>, Balance), ConvertMessageError> {
+		let network = Ethereum { chain_id };
+		let asset_hub_fee_asset: Asset = (Location::parent(), asset_hub_fee / 2).into();
```

Comment:
You can use full `asset_hub_fee` here. BuyExecution will only charge enough to cover weight of the XCM, the rest will be left in holding and will be available for transport fees.

Explicitly splitting it in half is more limiting as either one could fail when maybe one (exec or transpo) is bigger than half, but together still fit.

```suggestion
		let asset_hub_fee_asset: Asset = (Location::parent(), asset_hub_fee).into();
```